vagrant/ansible/Makefile

```diff
@@ -1,8 +1,8 @@
 INVENTORY := ./vagrant_ansible_inventory
 
-BACKEND ?= glusterfs
-
-ANSIBLE_EXTRA_VARS := -e 'backend=${BACKEND}'
+ifdef BACKEND
```

Comment:
I don't think you need to make any changes to the vagrant/Makefile and ansible/Makefile. We will be able to pass the backend variable by simply calling the playbook in the following manner

EXTRA_VARS="use_distro=centos7 backend=glusterfs" make

This will keep the Makefile clean.


I can add a document mentioning the variables which can be set. in a separate PR.

Reply:
I'm aware of how to set variables in Makefiles. But now suppose I want to support 2 things, like the OS and the backend, with 3 different OS and 3 backends. Will you need to create 16 Makefile targets (including the targets where none or just one of the components is specified), each explicitly setting all values in `EXTRA_VARS` ?

What will happen if we add more OSes, or backends, or options to enable/disable or configure components in the OS, samba or the backend ? the number of targets will grow exponentially pretty fast. Maybe the Makefile structure will be simple, but it will be unmaintainable and it will be easy to make mistakes.

Probably it would be possible to avoid the exponential grow, but at the cost of adding `ifdef` and other Makefile constructs, which is what you wanted to avoid. On the other hand, using just independent variables is easy to avoid the exponential grow.

In any case, the latest set of patches make use of `EXTRA_VARS`, so please take a look and tell me if there's any other thing that needs to be modified.